### DISCUSSION TITLE
Enhancing typing for modal component

### DISCUSSION THREAD

**Discussion by kaikno** [cloudscape-design/components#2542] [General]
Heyyo,

While looking at the demos, I noticed that you can provide a `closeAriaLabel` prop to the modal component, even if there isn't a `onDismiss` callback provided. This scenario, as shown in [this demo](https://cloudscape.design/examples/react/form-unsaved-changes.html), produces a modal with an accessible button that isnt actually very accessible cause you cant dismiss it. 

Perhaps the typing could be tightened up a little at this point. 😃

**Comment by pan-kot** [comment]
Hello,

Thank you for raising this question!

Although the `onDismiss` handler is optional in the API (because you can keep it undefined when the modal is not visible), the dismiss feature is required: the user should be able to close it by clicking on the dismiss button or pressing Escape.

The behaviour in the demo is a bug and I will address it. The [Communicating unsaved changes pattern](https://cloudscape.design/patterns/general/unsaved-changes/) already features the correct behaviour and correct code example.

**Comment by pan-kot** [comment]
The website behaviour has been fixed, closing the discussion now.
